File: CSharpStuff/Nathandelane.Math.Processor/Nathandelane.Math.Processor.Tokens/OpenPerenthesis.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Nathandelane.Math.Processor.Tokens
{
	public class OpenPerenthesis : IToken
	{
		#region Fields

		private TokenType _type;
		private TokenPrecedence _precedence;
		private string _value;

		#endregion

		#region Constructors

		public OpenPerenthesis()
		{
			_type = TokenType.Structure;
			_precedence = new TokenPrecedence(0);
			_value = "(";
		}

		#endregion

		#region IToken Members

		public TokenType Type
		{
			get { return _type; }
		}

		public TokenPrecedence Precedence
		{
			get
			{
				return _precedence;
			}
			set
			{
				_precedence = value;
			}
		}

		public string Value
		{
			get { return _value; }
		}

		#endregion
	}
}

File: JavaStuff/PersonalCalculator/personalcalculator/src/com/nathandelane/math/evaluation/tokens/OpenPerenthesis.java
package com.nathandelane.math.evaluation.tokens;

import java.util.regex.Pattern;

public class OpenPerenthesis extends TokenBase {
	
	public OpenPerenthesis() {
		super(TokenType.STRUCTURE, TokenPrecedence.OPEN_PERENTHESIS, "(");
		
		_pattern = Pattern.compile("^[(]{1}");
	}

}
